### ISSUE TITLE
Add support for JSONP and CORS for put's from browser

### ISSUE THREAD

**Issue by vikasaditya** [iron-io/issues#50]
Like amazon s3 now allows CORS, allow JSONP & CORS.
Would be great to gather data directly from users and process it latter.

**Comment by jpsilvashy** [comment]
@vikasaditya, you're able to do this already, using the postback that amazon sends to your server, you can then queue up a task. The postback from AWS will contain filesize, filetype, filename and filepath.

You can use those as parameters to kick off your worker to go get the image, process it, and store it somewhere new!

**Comment by vikasaditya** [comment]
@jpsilvashy 

I guess I was not clear.

We want to be able to add a message to your service directly from browser using JSONP or CORS.
Can we do this? 

We are using S3 as of now, but want to stop using that for this.

thanks.

**Comment by paddycarver** [comment]
Hey @vikasaditya,

This is something we've been discussing internally for quite some time. I'm looking into it, and hope we can have it included. Thanks for the suggestion. :)

**Comment by vikasaditya** [comment]
@paddyforan 

That would be great. 
There is a great first to the market opportunity for a cloud mq service that works with CORS or JSONP with open writes & restricted reads

**Comment by paddycarver** [comment]
While I'm inclined to agree, it's a tricky problem. OAuth2 security in insecure environments (i.e., client-side) is a problem without a real solution, so far. We've been talking internally about more granular controls on token permissions, but we don't really have anything to announce at this time.

It seems like the ability you're asking for is dependent on those new controls and the CORS requests. I'm looking into the CORS stuff, and may be able to get something released, but I can't make any promises as to if or when finer-grained permissions will be available for tokens.

**Comment by vikasaditya** [comment]
yes, CORS the way to go, s3 &  parse do it, which we have used, works great.
Thanks and will wait for your implementation.

**Comment by travischoma** [comment]
I see that the implementation has arrived this month: http://blog.iron.io/2014/03/ironio-enhances-web-and-mobile.html. Was there something done to address the security implications? The code example in the blog post seems to use the token client side?
